### ISSUE TITLE
Dict-like search space

### ISSUE THREAD

**Issue by davidfstein** [scikit-optimize/scikit-optimize#984]
I want to pass class weights in dict format as specified by the scikit api such as the following {0:10, 1:1}, {0:1, 1:1}, {0:1, 1:10}. However, I am getting TypeError: unhashable type: 'dict'. I see that in the past this was not supported, is it possible to pass dict search spaces like this now?

**Comment by kernc** [comment]
Can you show a code example of what you mean? Which "scikit api"?

Currently, you can pass [search spaces as `dimensions=`](https://scikit-optimize.github.io/stable/modules/generated/skopt.Optimizer.html#skopt.Optimizer).

**Comment by davidfstein** [comment]
Yes, see the example below. Meant to say Keras not scitkit, but the relevant documentation is as follows:

> class_weight: Optional dictionary mapping class indices (integers) to a weight (float) value, used for weighting the loss function (during training only). 

```
bayes_cv_tuner = BayesSearchCV(
        estimator = KerasClassifier(build_fn=create_model, verbose=1),
        scoring = 'average_precision',
        search_spaces = {
            'lr': (1e-5, 1),
            'nb_epoch': (80, 200),
            'batch_size': (8, 256),
            'class_weight': ({0:10, 1:1}, {0:1, 1:1}, {0:1, 1:10}),
        },
        cv = StratifiedKFold(
            n_splits=3,
            shuffle=True,
            random_state=42
        ),
            n_jobs = 80,
            n_points = 3,
            n_iter = 200,
            verbose = 0,
            refit = False,
            random_state = 1
    )
    result = bayes_cv_tuner.fit(X_train, y_train, callback=status_print)
```

**Comment by tsminhuang** [comment]
Hi, I had a similar issue and I found a workaround for that.
https://github.com/scikit-optimize/scikit-optimize/issues/681#issuecomment-781879964

**Comment by zeromh** [comment]
tsminhuang's fix worked for me, but I'd also love to see native support for dict-like search spaces!
